### ISSUE TITLE
Inspection `substr(..) used as index-based access` inspection is discouraged in PHP 7.4

### ISSUE THREAD

**Issue by Ayesh** [kalessil/phpinspectionsea#1454]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 4.0.1.1 |
| Language level | 7.4                       |

Current behaviour
--
When the `substr($str, 0, 1)` calls are used, the plugin suggests to use `$str[0]` as it offers some performance gains. While this indeed provides performance gains, this practice is deprecated in PHP 7.4: https://php.watch/versions/7.4/deprecated-string-array-index

Expected behaviour
--
Do not suggest this improvement, perhaps only for PHP >= 7.4.

Thanks a lot for the immense effort with this plugin Vladimir. I first learned about this pattern from this plugin, and it's rather unfortunate that PHP 7.4 deprecates this (although I can agree with the reasons behind them). 

When used in PHP 7.4, this trigger the following notice:

```
PHP message: PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in ... on line ...
```

**Comment by dryabov** [comment]
Curly braces are deprecated, so that `$str{0}` leads to a deprecated message, but `$str[0]` may be used safely.

**Comment by Ayesh** [comment]
Apologies, that is my bad. Thanks @dryabov for pointing that out.
